spec/order_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'order'
+
+describe "Order" do 
+
+  let (:menu){double :menu, dishes: {biriyani: 5.00, uk_curry: 6.00, cheeseburger: 4.00, pizza: 5.00, side_of_salmonella: 1.95}}
+  
+  before :each do
+    @subject = Order.new menu
+    allow(@subject).to receive(:send_sms).and_return true
+  end
+
+  context 'when created' do 
+    it 'has a selection hash' do
+      expect(@subject.selection).to be_a Hash
+    end
```

Comment:
Don't test for explicit types. If you need to change this type later on, this test will fail.